src/arcsf/models/config_class.py

```diff
@@ -0,0 +1,62 @@
+from arcsf.config.config_class import Config
+
+
+class ModelConfig(Config):
+
+    def __init__(
+        self,
+        model_id: str,
+        model_kwargs: dict,
+        trainer_kwargs: dict,
+    ) -> None:
```

Comment:
Thinking about this in terms of making it flexible for unlearning as well, maybe this could be modified to take a `trainer_type` as well, I guess as a `str` which can be used to load the relevant trainer/forgetter class (the draft forgetting PR has a class for each forget method which are all variants of the HuggingFace trainer with a different loss function and maybe extra arguments).

That could be done in the forgetting PR, though also worth thinking about config structure as fine-tuning and unlearning will likely need different hyperparameters. E.g. in the TOFU repo they have separate [model](https://github.com/locuslab/tofu/blob/main/config/model_config.yaml), [fine-tuning](https://github.com/locuslab/tofu/blob/main/config/finetune.yaml), and [forgetting](https://github.com/locuslab/tofu/blob/main/config/forget.yaml) configs.

Reply:
My inclination has been to have separate model and forgetting configs, with fine-tuning tied to the model config. The model config should then be re-used in forgetting (even if only to get the save destination path)

Reply:
I've modified the `load_trainer` fn to take a `trainer_type` string w/ associated dict to manage that string. Can talk about about it once the PR is ready for review

Reply:
Have also pegged the data collator class selection to this